wandb/sdk/launch/_project_spec.py

```diff
@@ -126,13 +151,25 @@ def _fetch_project_local(self, api: Api) -> None:
                 self.project_dir, run_info["git"]["remote"], run_info["git"]["commit"]
             )
             patch = utils.fetch_project_diff(self.uri, api)
-            if run_info.get("python"):
-                self._runtime = "python-{}".format(run_info["python"])
+
             if patch:
                 utils.apply_patch(patch, self.project_dir)
 
+            # For cases where the entry point wasn't checked into git
+            entry_point = run_info.get("codePath", run_info["program"])
+            if not os.path.exists(os.path.join(self.project_dir, entry_point)):
+                utils.download_entry_point(self.uri, api, entry_point, self.project_dir)
+                # if the entrypoint is downloaded and inserted into the project dir
+                # need to rebuild image with new code
+                self.build_image = True
+
+            # Download any frozen requirements
+            utils.download_wandb_python_deps(self.uri, api, self.project_dir)
```

Comment:
Tricky point here, if the frozen requirements.txt doesn't match the fronzen requirements for an existing image, we could run into issues.

Reply:
Actually, I don't think this will happen. I've already made it so the image is based on the source run/githuburl